tests/unit/browser/webengine/test_webengine_cookies.py

```diff
@@ -0,0 +1,73 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015-2019 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+from PyQt5.QtCore import QUrl
+from unittest.mock import MagicMock
+from qutebrowser.browser.webengine.cookies import _accept_cookie
+from qutebrowser.utils.urlmatch import UrlPattern
+
+
+def test_accept_cookie(config_stub):
+    """Test that _accept_cookie respects content.cookies.accept."""
+    request = MagicMock()
+
+    config_stub.set_str('content.cookies.accept', 'all')
+    assert _accept_cookie(request)
+
+    config_stub.set_str('content.cookies.accept', 'never')
+    assert not _accept_cookie(request)
```

Comment:
Please use [pytest parametrization](https://docs.pytest.org/en/latest/parametrize.html) to split this into individual subtests. Something like (untested):

```python3
import pytest

@pytest.mark.parametrize('value, third_party, accepted', [
    ('never', False, False),
    # ...
])
def test_accept_cookie(config_stub, filter_request, value, third_party, accepted):
    config_stub.val.content.cookies.accept = value
    filter_request.thirdParty = third_party
    assert cookies._accept_cookie(filter_request) == accepted
```